contracts/protocol/facets/ExchangeHandlerFacet.sol

```diff
@@ -757,14 +764,21 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
                         );
                     }
 
-                    // Make call only if code at address exists
-                    if (twin.tokenAddress.isContract()) {
-                        (success, result) = twin.tokenAddress.call(data);
+                    // Make call only if there is enough gas and code at address exists.
+                    // If not, skip the call and mark the transfer as failed
+                    if (gasleft() > reservedGas && twin.tokenAddress.isContract()) {
+                        bytes memory result;
+                        (success, result) = twin.tokenAddress.call{ gas: gasleft() - reservedGas }(data);
+
+                        success = success && (result.length == 0 || abi.decode(result, (bool)));
                     }
                 }
 
+                // Reduce minimum gas required for succesful execution
+                reservedGas -= SINGLE_TWIN_RESERVED_GAS;
+
                 // If token transfer failed
-                if (!success || (result.length > 0 && !abi.decode(result, (bool)))) {
+                if (!success) {
```

Comment:
why remove this? `|| (result.length > 0 && !abi.decode(result, (bool))`

Reply:
Wasn't removed, just turned around and moved to 773 (to solve stack too deep).

Reply:
nice